### PR TITLE
Add Category Filter

### PR DESCRIPTION
**Pull request by andrewdamelio**
relates to #2863
- Added ```<CategoryRow />``` component
- Built out the Category Filters
- added MarketsList store to hold market list meta data

**TODO (to close 2863)**
- wire up to load markets getter
- remove old unused category code
- minor cleanup/refactoring
- minor mobile fixes

### CODE REVIEW

File: packages/augur-ui/src/modules/common/constants.ts [hunk 1]
```diff
@@ -852,3 +852,5 @@ export const SCALAR_OUTCOMES = [
   },
 ];
 
+export const POPULAR_CATEGORIES = ['sports', 'politics', 'entertainment', 'finance', 'crypto'];
```
Comment (bthaile) [inline packages/augur-ui/src/modules/common/constants.ts:855]: Yes, these need to be hardcoded. Good place for them 👍

**Review by phoebemirman — COMMENTED** [review]
categories under popular header with equal number of markets aren't sorted alphabetically

File: packages/augur-ui/src/modules/markets-list/components/markets-view.tsx [hunk 2]
```diff
@@ -115,6 +118,7 @@ export default class MarketsView extends Component<
       includeInvalidMarkets,
     } = this.props;
     const { filter, sort, limit, offset } = this.state;
+    this.props.updateMarketsListSearching(true);
```
Comment (bthaile) [inline packages/augur-ui/src/modules/markets-list/components/markets-view.tsx]: `updateMarketsListSearching` isn't a good name. Seems like `clearCategoryMeta` or something like that, would be a better description

**Comment by bthaile** [comment]
awesome, only one real comment, just for code clarity

**Review by bthaile — APPROVED** [review]
lgtm, Phoebe is also reviewing

File: packages/augur-ui/src/modules/markets-list/components/markets-view.tsx (hunk 2 above)
Comment (andrewdamelio) [reply packages/augur-ui/src/modules/markets-list/components/markets-view.tsx]: 👍

**Comment by andrewdamelio** [comment]
> categories under popular header with equal number of markets aren't sorted alphabetically

Will do I missed that detail 👍

File: packages/augur-ui/src/modules/app/components/inner-nav/category-filters.styles.less [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  > .CategoriesGroup:first-of-type {
+    margin: 0 @size-40 @size-40;
+
+    @media @breakpoint-mobile {
+      margin: @size-16;
+    }
+
+    > div {
+      margin: 0;
+      height: 14px;
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/app/components/inner-nav/category-filters.styles.less]: you should use the constant here

File: packages/augur-ui/src/modules/app/components/inner-nav/category-filters.styles.less [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+.SelectedCategories_back {
+  margin: @size-8 @size-32;
+  user-select: none;
+  cursor: pointer;
+
+  > svg {
+    fill: @color-secondary-text;
+    height: 10px;
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/app/components/inner-nav/category-filters.styles.less]: a few places in this file that should have constants

File: packages/augur-ui/src/modules/app/components/inner-nav/category-filters.tsx [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+  renderSelectedCategories() {
+    return (
+      <div className={Styles.SelectedCategories}>
+        {this.state.selectedCategories
+          .filter(categories => categories !== this.state.selectedCategory)
+          .map((category, idx) => {
+            return (
+              <div
+                key={idx}
+                className={Styles.SelectedCategories_back}
```
Comment (phoebemirman) [inline packages/augur-ui/src/modules/app/components/inner-nav/category-filters.tsx]: not sure if this needs its own classname. and also thought we are moving away from underscore classnames

Comment (andrewdamelio) [reply packages/augur-ui/src/modules/app/components/inner-nav/category-filters.tsx]: 👍 good call I will remove the class name completely

**Review by phoebemirman — COMMENTED** [review]
Screen Shot 2019-08-22 at 1 35 47 PM

getting this error sometimes, switched to in reporting at one point



>Been trying but can reproduce - will keep an eye on this while I work on the 2nd part of this feature.
> andrewdamelio

**Review by phoebemirman — APPROVED** [review]
looks good, few comments

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). 🏆

 This link will always point to this PR's latest changes: https://peek.run/r9zhxrz

**Comment by bthaile** [comment]
Why shouldn't categories be alphabetized if they have the same number of markets. I might be rem incorrectly.  If not maybe for the popular categories they always have the same order. easily changed in the future.

**Comment by andrewdamelio** [comment]
@bthaile working on the feedback this AM -- will merge after feedback fixes get merged

**Pull request merged**